irohad/main/impl/ordering_init.cpp

```diff
@@ -6,13 +6,24 @@
 #include "main/impl/ordering_init.hpp"
 #include "ametsuchi/ordering_service_persistent_state.hpp"
 #include "interfaces/common_objects/peer.hpp"
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/iroha_internal/block.hpp"
 
 namespace iroha {
   namespace network {
     auto OrderingInit::createGate(
         std::shared_ptr<OrderingGateTransport> transport,
         std::shared_ptr<ametsuchi::BlockQuery> block_query) {
-      auto height = block_query->getTopBlocks(1).as_blocking().last()->height();
+      shared_model::interface::types::HeightType height;
+      block_query->getTopBlock().match(
+          [&height](
+              expected::Value<std::shared_ptr<shared_model::interface::Block>>
+                  block) { height = block.value->height(); },
```

Comment:
To avoid declaring height out of the scope, we can write createGate like this:
<img width="531" alt="image" src="https://user-images.githubusercontent.com/14168731/41531597-7ffbd7a8-72fc-11e8-81b6-262dbf49be44.png">